frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts

```diff
@@ -0,0 +1,681 @@
+                ...rawCommandRegistrations,
+                custom_dashboards: {
+                    key: 'custom_dashboards',
+                    resolver: dashboards.map((dashboard: DashboardType) => ({
+                        key: `dashboard_${dashboard.id}`,
+                        icon: LineChartOutlined,
+                        display: `Go to Dashboard ${dashboard.name}`,
+                        executor: () => {
+                            const { push } = router.actions
+                            push(`/dashboard/${dashboard.id}`)
+                        },
+                    })),
+                    scope: GLOBAL_COMMAND_SCOPE,
+                },
+            }),
+        ],
+        regexpCommandPairs: [
+            (selectors) => [selectors.commandRegistrations],
+            (commandRegistrations: CommandRegistrations) => {
+                const array: RegExpCommandPairs = []
+                for (const command of Object.values(commandRegistrations)) {
+                    if (command.prefixes)
+                        array.push([new RegExp(`^\\s*(${command.prefixes.join('|')})(?:\\s+(.*)|$)`, 'i'), command])
+                    else array.push([null, command])
+                }
+                return array
+            },
+        ],
+        commandSearchResults: [
+            (selectors) => [selectors.regexpCommandPairs, selectors.input, selectors.activeFlow, selectors.isSqueak],
+            (
+                regexpCommandPairs: RegExpCommandPairs,
+                argument: string,
+                activeFlow: CommandFlow | null,
+                isSqueak: boolean
+            ) => {
+                if (isSqueak) return []
+                if (activeFlow) return resolveCommand(activeFlow, argument)
+                let directResults: CommandResult[] = []
+                let prefixedResults: CommandResult[] = []
+                for (const [regexp, command] of regexpCommandPairs) {
+                    if (regexp) {
+                        const match = argument.match(regexp)
+                        if (match && match[1]) {
+                            prefixedResults = [...prefixedResults, ...resolveCommand(command, match[2], match[1])]
+                        }
+                    }
+                    directResults = [...directResults, ...resolveCommand(command, argument)]
+                }
+                const allResults = directResults.concat(prefixedResults)
+                let fusableResults: CommandResult[] = []
+                let guaranteedResults: CommandResult[] = []
+                for (const result of allResults) {
+                    if (result.guarantee) guaranteedResults.push(result)
+                    else fusableResults.push(result)
+                }
+                fusableResults = _.uniqBy(fusableResults, 'display')
+                guaranteedResults = _.uniqBy(guaranteedResults, 'display')
```

Comment:
could we replace lodash with some local `lib` utilities?

Reply:
We probably could, but gotta say lodash is convenient…

Reply:
So is jQuery :D.

Trolling aside, I think in the worst case we should at least use the `lodash/uniqBy` type tree-shakeable imports, though having our own tiny helper functions is IMO the best solution. The less external dependencies the better.